proposals/NNNN-library-evolution.md

```diff
@@ -0,0 +1,369 @@
+
+Binaries compiled without this mode should not be declared as ABI-stable. While they should be stable even without library evolution mode turned on, doing this will not be a supported configuration.
+
+## Detailed design
+
+### Library evolution mode
+
+A new command-line argument, `-enable-library-evolution`, will enable this new mode.
+
+Turning on library evolution mode will have the following effects on source code:
+
+- The default behavior for enums will change to be non-frozen. This is the only change visible to _users_ of the library, who will need to use the `@unknown default:` technique described in SE-0192.
+- To ensure that all fields are initialized, inlinable `init`s of the type must delegate to a non-inlinable `init`.
+
+### `@frozen` on `struct` types
+
+When a library author is certain that there will never be a need to add fields to a struct in future, they may mark that type as `@frozen`. 
+
+This will allow the compiler to optimize away at compile time some calls that would otherwise need to be made at runtime (for example, it might access fields directly without the indirection).
+
+When compiling with binary stability mode on, a struct can be marked `@frozen` as long as it meets all of the following conditions:
+
+- The struct is *ABI-public* (see [SE-0193][]), i.e. `public` or marked `@usableFromInline`.
+- Every class, enum, struct, protocol, or typealias mentioned in the types of the struct's fields is ABI-public.
+- No fields have observing accessors (`willSet` or `didSet`).
+- If a field has an initial value, the expression computing the initial value does not reference any types or functions that are not ABI-public.
+
+```swift
+@frozen
+public struct Point {
+  public var x: Double
+  public var y: Double
+
+  public init(_ x: Double, _ y: Double) {
+    self.x = x
+    self.y = y
+  }
+}
+```
+
+This affects what changes to the struct's fields affect the ABI of the containing library:
+
+| Change | Normal struct | `@frozen` struct
+|---|:---:|:---:
+| Adding fields | Allowed | **Affects ABI**
+| Reordering fields | Allowed | **Affects ABI**
+| Removing ABI-public fields | Affects ABI | Affects ABI
+| Removing non-ABI-public fields | Allowed | **Affects ABI**
+| Changing the type of an ABI-public field | Affects ABI | Affects ABI
+| Changing the type of a non-ABI-public field | Allowed | **Affects ABI**
+| Changing a stored instance property to computed | Allowed | **Affects ABI**
+| Changing a computed instance property to stored | Allowed | **Affects ABI**
+| Changing the access of a non-ABI-public field | Allowed | Allowed
+| Marking an `internal` field as `@usableFromInline` | Allowed | Allowed
+| Changing an `internal` ABI-public field to be `public` | Allowed | Allowed
```

Comment:
What's with the inconsistent bolding?

Reply:
Bold indicates inconsistency between frozen and unfrozen.

Reply:
And how am I as a reader supposed to know that?